app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1031,6 +1066,12 @@ class BrowserTabViewModel(
         command.value = LaunchTabSwitcher
     }
 
+    private fun canFireproofWebsite(): Boolean {
+        val domain = site?.uri?.host ?: return false
+        val fireproofWebsites = fireproofWebsiteState.value
+        return fireproofWebsites?.all { it.domain != domain } ?: true
```

Comment:
I explored that option, but I didn't want to refactor the methods that are using this logic to introduce `viewModelScope.launch` to call the database. There are multiple methods that can trigger this logic and not all of them are easy to refactor without further testing. So I went for this approach for simplicity.

`Iterable<T>.all` will stop when the first item matches the criteria, I don't think we will notice the difference, and it can be revisited at some point if we need to improve performance.